pandas/core/arrays/boolean.py

```diff
@@ -565,33 +568,40 @@ def logical_method(self, other):
                 # Rely on pandas to unbox and dispatch to us.
                 return NotImplemented
 
+            assert op.__name__ in {"or_", "ror_", "and_", "rand_", "xor", "rxor"}
             other = lib.item_from_zerodim(other)
+            other_is_booleanarray = isinstance(other, BooleanArray)
+            other_is_scalar = lib.is_scalar(other)
             mask = None
 
-            if isinstance(other, BooleanArray):
+            if other_is_booleanarray:
                 other, mask = other._data, other._mask
             elif is_list_like(other):
                 other = np.asarray(other, dtype="bool")
                 if other.ndim > 1:
                     raise NotImplementedError(
                         "can only perform ops with 1-d structures"
                     )
-                if len(self) != len(other):
-                    raise ValueError("Lengths must match to compare")
                 other, mask = coerce_to_array(other, copy=False)
+            elif isinstance(other, np.bool_):
+                other = other.item()
```

Comment:
this is to convert to a python bool?  why not just `bool(other)`?  `item` i usually think of as being an array method

Reply:
`item` is the general method to get a python scalar (here we of course know we want a bool).

But Tom, why is it exactly needed to convert this? I would think the numpy operations later on work fine with a numpy scalar as well?

Reply:
IIRC, we do things like `if right is False` or `if right is True`, which will fail for numpy booleans. I don't want to have to worry about checking both, so easier to convert here.